src/bp/nlu/engine/engine/index.ts

```diff
@@ -55,11 +55,17 @@ export default class Engine implements Engine {
 
   constructor(opt?: Partial<EngineOptions>) {
     const options: EngineOptions = { ...DEFAULT_ENGINE_OPTIONS, ...opt }
+
     this.modelsById = new LRUCache({
-      max: options.maxCacheSize,
+      max: options.maxCacheSize < 0 ? undefined : options.maxCacheSize,
```

Comment:
I'm clearly being a jerk right now but can you extract the parsing and validation of cacheSize in a 
a nlu common util

Reply:
no need to, all I do now is `Math.abs()`